documentation/src/docs/reader-group-design.md

```diff
@@ -9,85 +9,90 @@ You may obtain a copy of the License at
+1. The new Reader can take over a segment by recording that it is doing so in the shared state.
+1. The new Reader can start reading from the position it read from the shared state for the segment it picked up.
 
-There are no races between multiple readers coming online concurrently because only one of them can successfully claim ownership of any given segment.
+There are no races between multiple Readers coming online concurrently because only one of them can successfully claim ownership of any given segment.
 
 ## Segments merging
-1. When a reader comes to the end of its segment it records this information in the shared state.
-1. When all of the segments that are getting merged together are completed a reader may claim ownership of the following segment.
+1. When a Reader comes to the end of its segment it records this information in the shared state.
+1. When all of the segments that are getting merged together are completed, a Reader may claim ownership of the following segment.
+
+There is no ambiguity as to who the owner is, because it is stored in the shared state. There is no risk of a segment being ignored because every Reader can see the available segments by looking at the shared state and claim them.
+
+## Reader going offline
+1. When a Reader dies, the `readerOffline(readerId, position) api` method will be invoked either by the Reader itself in a graceful shutdown (internally to the close method) or via a "liveness detector". In either case the Reader's last position is written to the state.
+
+1. If a null `Position` is sent then the last checkpointed position will be written to the state.
+1. This is used by the newer Readers when they take ownership of the segment(s) that were read by the older/offline reader.
+1. Any Reader can decide to take over one or more of the segments owned by the old Reader from where it left off by recording their intention to do so in the state object.
+1. Once the state has been updated by the new Reader, it is considered the owner of the segment and can read from it.
 
-There is no ambiguity as to who the owner is, because it is stored in the shared state. There is no risk of a segment being forgotten because any reader can see which segments are available by looking at the shared state and claim them.
 
-## Reader going offline.
-1. When a reader dies the readerOffline() method will be invoked either by the reader itself in a graceful shutdown (internally to the close method) or via a liveness detector. In either case the reader's last position is passed in.
-1. The last position is written to the state.
-1. Other readers will see this when they update their local state.
-1. Any of them can decide to take over one or more of the segments owned by the old reader from where it left off by recording their intention to do so in the state object.
-1. Once the state has been updated the new reader is considered the owner of the segment and can read from it at will.
 
```

Comment:
Here we can introduce another section called for example `Other Considerations on Reader Groups` to distinguish the two next subsections (which are not operations) from the previous ones.

Reply:
Accepted suggestion